core/2d/CCDrawNode.cpp

```diff
@@ -369,8 +369,8 @@ void DrawNode::drawCircle(const Vec2& center,
     }
     if (_lineWidth > minThickness)
     {
-        auto scaleFactor = AX_CONTENT_SCALE_FACTOR() * 2 ;
-        drawPolygon(vertices, segments, Color4B(1.0f, 0.0f, 0.0f, 1.0f), _lineWidth / scaleFactor, color);
+        static GLView* glView = _director->getOpenGLView();
+        drawPolygon(vertices, segments, Color4B(1.0f, 0.0f, 0.0f, 1.0f), _lineWidth / (glView->getScaleX() + glView->getScaleY()), color);
```

Comment:
I think, you can use _lineWidth directly, don't use any scale from glview

Reply:
I have improve the tester on cpp_test for better see the "improvment".
Its an own PR.

Reply:
> I think, you can use _lineWidth directly, don't use any scale from glview

It is not working correct only using the _lineWidth without scaling it.

Reply:
In your test case, you set the border so much large, right?

Reply:
> In your test case, you set the border so much large, right?

Yes. its a test case to see it works also with a "big" _linewith too.
Maybe not a real situation...but a worst case which have to test.

Reply:
Your code based on cpp-tests with frameSize(960, 640) design size(480, 320), it's very specific, when you change both of them to 1280x720, then looks good without scale, and you scale code looks a little magic.

Reply:
![image](https://user-images.githubusercontent.com/6977319/218044483-98fba72f-5ede-408c-bb17-e7f690008da5.png)

Reply:
Its magic ;)

Reply:
![image](https://user-images.githubusercontent.com/6977319/218045459-b8701e36-d64f-4eaa-ae1f-1cc553f16777.png)

This maybe better.

Reply:
```
        GLView* glView = _director->getOpenGLView();
        auto borderScale = (glView->getScaleX() + glView->getScaleY());
        drawPolygon(vertices, segments, Color4B(1.0f, 0.0f, 0.0f, 1.0f), _lineWidth / borderScale , color);
```

Only this is working with different "window settings like:
```
static Size resourceSize(960, 640);
static Size resourceSize(1280, 720);

static Size designSize(480, 320);
static Size designSize(400, 320);
```

Reply:
@halx99 
I change it like this and PR will be merged?
```
GLView* glView = _director->getOpenGLView();
        auto borderScale = (glView->getScaleX() + glView->getScaleY());
        drawPolygon(vertices, segments, Color4B(1.0f, 0.0f, 0.0f, 1.0f), _lineWidth / borderScale , color);
```

Reply:
Yes